pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -416,14 +416,17 @@ func nodesWherePreemptionMightHelp(nodes []*framework.NodeInfo, m framework.Node
 	var potentialNodes []*framework.NodeInfo
 	nodeStatuses := make(framework.NodeToStatusMap)
 	for _, node := range nodes {
-		name := node.Node().Name
+		nodeName := node.Node().Name
 		// We rely on the status by each plugin - 'Unschedulable' or 'UnschedulableAndUnresolvable'
-		// to determine whether preemption may help or not on the node.
-		if m[name].Code() == framework.UnschedulableAndUnresolvable {
-			nodeStatuses[node.Node().Name] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "Preemption is not helpful for scheduling")
-			continue
+		// to determine whether preemption may help or not on the node. We first check that the entry
+		// is in the map, as a missing entry implies UnschedulableAndUnresolvable.
+		if status, ok := m[nodeName]; ok {
+			if status.Code() == framework.UnschedulableAndUnresolvable {
+				nodeStatuses[nodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "Preemption is not helpful for scheduling")
```

Comment:
Shouldn't we set `framework.UnschedulableAndUnresolvable` for nodes that are not in the map also?
```
	for _, node := range nodes {
		nodeName := node.Node().Name
		status, ok := m[nodeName]
		if !ok || status.Code() == framework.UnschedulableAndUnresolvable {
			nodeStatuses[nodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "Preemption is not helpful for scheduling")
			continue
		}
		potentialNodes = append(potentialNodes, node)
	}
```

Reply:
> Thousands of identical messages about all-but-one node rejected by prefilter are not useful and weren't emitted prior to the latest round of patch releases.

OK, sgtm. 

We did set `UnschedulableAndUnresolvable` for every nodes rejected by prefilter https://github.com/kubernetes/kubernetes/blob/be4afb9ef90b19ccb6f7e595cbdb247e088b2347/pkg/scheduler/schedule_one.go#L495, so I was a little concerned about inconsistency.

Reply:
> Please open an issue for improving efficiency of unschedulable Daemonsets.

Oh, actually, after this PR, the only case where we would fill up this second map is when all the Filter plugins returned UnschedulableAndUnresolvable when PreFilter didn't. I think this is generally uncommon or a pathological case. One example would be a nodeSelector for a label that doesn't exist in any node.

We could think of improving those scenarios, but maybe it's not that important.

Reply:
> Shouldn't we set `framework.UnschedulableAndUnresolvable` for nodes that are not in the map also?
> 
> ```
> 	for _, node := range nodes {
> 		nodeName := node.Node().Name
> 		status, ok := m[nodeName]
> 		if !ok || status.Code() == framework.UnschedulableAndUnresolvable {
> 			nodeStatuses[nodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "Preemption is not helpful for scheduling")
> 			continue
> 		}
> 		potentialNodes = append(potentialNodes, node)
> 	}
> ```

As there wasn't a performance issue here previously, and the diagnostic message [becomes inaccurate](https://github.com/kubernetes/kubernetes/pull/125197#discussion_r1621987135) (counts number of nodes wrongly), I prefer the option @AxeZhan suggested ^. We can then update the calculation diagnostic message to be more efficient in a subsequent commit which will not be cherry picked. Wdyt?

Reply:
This logic will immediately become problematic whenever there are multiple daemonset pods that don't fit.
But (1) that's already a pathological case and (2) it doesn't look like it would be a regression from today's behavior, because we were always populating this map.

But, to be sure, can you confirm my statement in (2) before #124559

Reply:
Uhm... I think it might actually have been worse in #124559, because a node absent from the map was considered just `Unschedulable`?

Reply:
> (2) it doesn't look like it would be a regression from today's behavior, because we were always populating this map.

this is my understanding as well.